spec/requests/case_log_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe CaseLogsController, type: :request do
+  describe "POST #create" do
+    let(:tenant_code) { "T365" }
+    let(:tenant_age) { 35 }
+    let(:property_postcode) { "SE11 6TY" }
+    let(:api_username) { "test_user" }
+    let(:api_password) { "test_password" }
+    let(:basic_credentials) do
+      ActionController::HttpAuthentication::Basic
+                          .encode_credentials(api_username, api_password)
+    end
+
+    let(:headers) do
+      {
+        "Content-Type" => "application/json",
+        "Accept" => "application/json",
+        "Authorization" => basic_credentials,
+      }
+    end
+
+    let(:params) do
+      {
+        "tenant_code": tenant_code,
+        "tenant_age": 35,
+        "property_postcode": property_postcode,
+      }
+    end
+
+    before do
+      allow(ENV).to receive(:[])
```

Comment:
What does this line do? I can tell what 33 and 34 do but curious about this one 🙂

Reply:
If you don't set it then having 33 and 34 means Rails thinks those are the _only_ arguments ENV is allowed to receive for `:[]` so any other ENV vars that are looked up stop working. That means you can't for example put a `binding.pry` in your test because that relies on `ENV["DISABLE_PRY"]`